lib/models/addon.js

```diff
@@ -165,6 +165,86 @@ function warn(message) {
   @param {Project} project The current project (deprecated)
 */
 let addonProto = {
+  /**
+    The name of this addon.
+
+    @public
+    @final
+    @type String
+    @property name
+  */
+
+  /**
+    The root directory where this addon is located.
+
+    @public
+    @final
+    @type String
+    @property root
+  */
+
+  /**
+    The host app instance.
+
+    **Note**: this property will only be present on addons that are a direct dependency
+    of the application itself, not of other addons. It is also only available after the
+    base `included()` hook has been run.
```

Comment:
This last bit is not correct, `this.app` is set _before_ `setupPreprocessorRegistry` is called (which is also before `included` is called):

https://github.com/ember-cli/ember-cli/blob/6e84ea4da023960f89404a781375223e6c833609/lib/broccoli/ember-app.js#L143

I think the main point to make in the docs would be that it is _not_ available in `init`, and that it is present before `setupPreprocessorRegistry` / `included` are called.

Reply:
Oops! For some reason I'd convinced myself it was the base `included()` hook that set that.

Reply:
yep, it used to be